packages/mwp-app-render/src/components/connectedWithMatchMedia.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { createStructuredSelector, createSelector } from 'reselect';
+import withMatchMedia from 'meetup-web-components/lib/utils/components/withMatchMedia';
+
+/**
+ * getConfig
+ * @param  {Object} state
+ * @return {Object} the config from state
+ */
+const getConfig = state => state.config || {};
+
+/**
+ * Checks if object is empty
+ * @param  {Object} object
+ * @return {Boolean} whether object is empty
+ */
+const isEmpty = object => object === undefined || Object.keys(object).length === 0;
+
+/**
+ * @param {Object} device - object containing device information
+ * @returns {Object} - default media object for `withMatchMedia`
+ */
+const getMediaOrDefault = (media, device) => {
+	if (!isEmpty(media)) {
+		return media;
+	}
+	if ( !isEmpty(device) ) {
+		const {isMobile, isTablet, isDesktop} = device;
+		const isTabletOrDesktop = isTablet || isDesktop;
+		return {
+			isAtSmallUp: Boolean(isMobile || isTabletOrDesktop),
+			isAtMediumUp: Boolean(isTabletOrDesktop),
+			isAtLargeUp: Boolean(isDesktop),
+		}
+	}
+	return {};
+};
+
+const mapStateToProps = createStructuredSelector({
+	device: createSelector(getConfig, config => (config.device || {})),
+});
+
+/**
+ * connectedWithMatchMediaComponent
+ * HOC that provides a default based on `device` in state.config
+ * if withMatchMedia has not yet detected the media
+ *
+ * `device` prop is provided from redux state in config.
+ * `media` prop is provided to the wrapped component in render().
+ *
+ * @returns {React.element}
+ */
+const connectedWithMatchMedia = WrappedComponent => {
+
+	/**
+	 * @module ConnectedWithMatchMedia
+	 */
+
+	class ConnectedWithMatchMedia extends React.Component {
+
+		render() {
+			const {device, media: currentMedia } = this.props;
+			const media = getMediaOrDefault(currentMedia, device);
+			return <WrappedComponent {...this.props} media={media}/>;
+		}
+	};
+	const wrappedComponentName =
+		WrappedComponent.displayName || WrappedComponent.name || 'Component';
+
+	ConnectedWithMatchMedia.WrappedComponent = WrappedComponent;
+	ConnectedWithMatchMedia.displayName = `ConnectedWithMatchMedia(${wrappedComponentName})`;
+	
+	/* connect to redux state */
```

Comment:
just a little thing, but since `connect` is a HOC with well-understood behavior, I don't think it needs a comment, and `return withMatchMedia(ConnectedComponent)` is also pretty legible as a HOC, so `/* return connected wrapped component withMatchMedia */` is probably unnecessary.

Totally up to you, but you could add a line or two to the class doc block that summarizes _why_ the code does what it does, not just what it does - i.e. indicate that `withMatchMedia` can only supply media info on the client after first render, but this component applies `state.config.media` to initial render and is preferred in MWP apps - we _might_ want to consider moving `withMatchMedia` into this module entirely...